### PR TITLE
Support new AbstractFFTs.AdjointStyle trait for FFT and DCT plans

### PR DESCRIPTION
**Pull request by gaurav-arya**
Supports the new `AbstractFFTs.AdjointStyle` trait. ~This is a draft PR because https://github.com/JuliaMath/AbstractFFTs.jl/pull/78 and https://github.com/JuliaMath/AbstractFFTs.jl/pull/109 are not merged yet, which this PR relies on.~

### CODE REVIEW

**Comment by vpuri3** [comment]
@gaurav-arya where is `testbackend.jl`?

**Comment by gaurav-arya** [comment]
I think you can remove that: it was one attempt at reusing AbstractFFTs' tests here but we haven't yet agreed on a good solution, see https://github.com/JuliaMath/AbstractFFTs.jl/pull/78.

If you'd like to push this through perhaps if you can think of a minimal self contained test here to test the traits, we could sidestep the problem and get this merged (note I don't have merge rights, so ultimately the decision should be verified with someone who does).

**Comment by gaurav-arya** [comment]
This is ready!

File: src/dct.jl [hunk 1]
```diff
@@ -171,3 +171,5 @@ end
     mul!(Array{T}(undef, p.plan.osz), p, copy(x)) # need copy to preserve input
 
 *(p::DCTPlan{T,K,true}, x::StridedArray{T}) where {T,K} = mul!(x, p, x)
+
+AbstractFFTs.AdjointStyle(::DCTPlan) = AbstractFFTs.UnitaryAdjointStyle()
```
Comment (stevengj) [inline src/dct.jl:175]: What will we do for the rest of the r2r plans?  e.g. the transpose of REDFT01 is REDFT10 + diagonal scaling.

Comment (vpuri3) [reply src/dct.jl:175]: we're leaving r2r plans for a future PR. it can get pretty complicated with fishing multiple kinds of plans

Comment (gaurav-arya) [reply src/dct.jl:175]: Specifically, we can define new adjoint styles here for those cases -- that's planned for a separate PR 🙂

**Comment by gaurav-arya** [comment]
Bump

**Comment by stevengj** [comment]
LGTM.

**Comment by gaurav-arya** [comment]
@stevengj could we get a release for this PR? It would be helpful for including FFTW's run of the test suite in downstream tests for https://github.com/JuliaMath/AbstractFFTs.jl/pull/118

**Pull request merged**